src/main/java/mods/railcraft/common/blocks/machine/TileMachineItem.java

```diff
@@ -16,7 +16,9 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 
-public abstract class TileMachineItem extends TileMachineBase implements IInventory {
+public abstract class TileMachineItem extends TileMachineBase 
+// FIXME REMOVE and replace with capabilities
+implements IInventory {
```

Comment:
Then forge is short sighted, IItemHandler is too restrictive for many use cases.